packages/tpetra/core/src/Tpetra_Import_def.hpp

```diff
@@ -297,96 +318,88 @@ namespace Tpetra {
       } catch (Teuchos::Exceptions::InvalidParameter&) {}
     }
     debug_ = debug;
+    ArrayView<const GO> sourceGIDs = source->getNodeElementList ();
+    ArrayView<const GO> targetGIDs = target->getNodeElementList ();
+    const size_type numSrcGids = sourceGIDs.size ();
```

Comment:
It's better not to call getNodeElementList if you don't actually use the GIDs.  If you do use them, it's OK, but still preferred to have special cases for contiguous Maps.  I won't insist on this, though.